addons/binding/org.openhab.binding.network/src/main/java/org/openhab/binding/network/service/NetworkUtils.java

```diff
@@ -146,25 +178,125 @@ public static final long inetAddress2Long(InetAddress addr) {
         return (inetAddress2Int(addr) & 0xFFFFFFFFL);
     }
 
-    public static boolean nativePing(String hostname, int port, int timeout)
-            throws InvalidConfigurationException, IOException, InterruptedException {
+    /**
+     * Try to establish a tcp connection to the given port. Returns false if a timeout occurred
+     * or the connection was denied.
+     *
+     * @param host The IP or hostname
+     * @param port The tcp port. Must be not 0.
+     * @param timeout Timeout in ms
+     * @param logger A slf4j logger instance to log IOException
+     * @return
+     */
+    public static boolean servicePing(String host, int port, int timeout, Logger logger) {
```

Comment:
Why pass the logger you can and imho should use a static logger in this case. Otherwise you lose the context where this logging came from.